tools/cmake/components.cmake

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+list(APPEND MBED_COMPONENTS_LIST mbed-ble mbed-cellular mbed-nfc mbed-nanostack mbed-netsockets mbed-wifi)
+
+# Add libraries to specified components
+# This can be used to link additional libraries that are required
+# ARGN is a list of libraries to add to mbed-<component>
+function(mbed_add_component_libraries component)
+    # we use global symbols, using upper case
+    string(TOUPPER ${component} component_upper)
+    if(component IN_LIST MBED_COMPONENTS_LIST)
+        set(${component_upper}_EXTRA_LIBRARIES "${${component}_EXTRA_LIBRARIES};{ARGN}" CACHE INTERNAL "")
```

Comment:
Can we pass the name of the extra library instead of the building it from `component`?

Reply:
> I am even wondering if the function is needed at all...

I am also wondering why we need this function. If there's a strong justification for this we should document it somewhere, preferably in the commit message.

Another issue I'm having with this code is that we have now introduced this "component" terminology, which is confusing, as CMake does already have a "components" concept. I think the overloading of the word "component" here will be confusing for future developers.

Reply:
I can fix wording, first lets agree if we want a variable to be accessible or function/macro.

We need to have a list of extra modules accessible in the tree. This can be done via cache variable or property. 

If it's cache, a user would need to append to it (I wish there is list append or similar for a cache, I only was able to   make ti work this way). Similar as it is done in the function:
```
set(MBED_BLE_EXTRA_LIBRARIES "${MBED_BLE_EXTRA_LIBRARIES};target-extra-libs" CACHE INTERNAL "")
```

Is that straightforward to a user  to do this ? If you both agree, I'll switch back.

Reply:
Ok I think I understand that the point of this function is to be a sort of utility function that helps to add values to a cache variable. The use case is as follows:
If we do not have the function, for a tree like:
```
+ foo/
    + CMakeLists.txt
    + bar/
        + CMakeLists.txt
```
If `foo/CMakeLists.txt` creates the cache variable `XXX_EXTRA_LIBRARIES` and `bar/CMakeLists.txt` then wants to append to it it needs to do:
```
set(XXX_EXTRA_LIBRARIES
    ${XXX_EXTRA_LIBRARIES}
    <OTHER_CONNECTIVITY_DRIVER_CMAKE_TARGET>
    CACHE INTERNAL ""
)
```
This does not seem very elegant and I see how the function helps. I do not have a strong opinion on whether to keep the function or not. However, I would request for it to be generic as in it's just a function to append values to a cache variable. It should therefore have as input the name of the variable and the values to add to it.

Reply:
However, not having the function gives the possibility to easily remove libraries added by parent directories. In the example above, if `bar/CMakeLists.txt` wanted to remove one of the libraries added it could just overwrite the cache variable `XXX_EXTRA_LIBRARIES` with `set(XXX_EXTRA_LIBRARIES
 <NOTHING_OR_ONLY_THE_DRIVERS_NEEDED> CACHE INTERNAL "")`.

Reply:
I'll switch it back.

Reply:
Fixed